kclvm/vfs/src/vfs.rs

```diff
@@ -0,0 +1,143 @@
+use anyhow::anyhow;
+use anyhow::Ok;
+use anyhow::Result;
+use parking_lot::RwLock;
+
+use std::fs;
+use std::panic;
+use std::panic::AssertUnwindSafe;
+use std::{
+    collections::HashMap,
+    hash::Hash,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+use crate::sourcemap::SourceMapVfs;
+use lazy_static::lazy_static;
+use std::sync::Mutex;
+
+pub trait VFS {
+    // TODO: More actions to be get SourceFile directly
+    fn write(&self, path: String, contents: Option<Vec<u8>>) -> Result<()>;
```

Comment:
Considering a `AsRef<Path>` or `AsRef<Utf8Path>` parameter for the `path` variable.